environments/datasets.py

```diff
@@ -0,0 +1,108 @@
+        """Constructor method
+
+        Args:
+            environment (Environment): Instance belonging to Environment abstract base class.
+            length (int): Length of the dataset.
+            number_of_frames (int): Total duration of video (in frames).
+            delta_time (float): Frame interval of generated data (in seconds).
+            number_of_rollouts (int): Number of rollouts to generate.
+            img_size (int): Size of the frames (in pixels).
+            color (bool): Whether to have colored or grayscale frames.
+            noise_std (float): Standard deviation of the gaussian noise source, no noise for noise_std=0.
+            radius_bound (float, float): Radius lower and upper bound of the phase state sampling.
+                Init phase states will be sampled from a circle (q, p) of radius
+                r ~ U(radius_bound[0], radius_bound[1]) https://arxiv.org/pdf/1909.13789.pdf (Sec. 4)
+            world_size (float) Spatial extent of the window where the rendering is taking place (in meters).
+            seed (int): Seed for reproducibility.
+            data_mean (float): Data mean for standaritzation. Defaults to 0.5.
+            data_std (float): Data std for standaritzation. Defaults to 0.5.
+        """
+        self.environment = environment
+        self.length = length
+        self.data_mean = data_mean
+        self.data_std = data_std
+        self.number_of_frames = number_of_frames
+        self.delta_time = delta_time
+        self.number_of_rollouts = number_of_rollouts
+        self.img_size = img_size
+        self.noise_std = noise_std
+        self.radius_bound = radius_bound
+        self.world_size = world_size
+        self.seed = seed
+
+    def __len__(self):
+        """Get dataset length
+
+        Returns:
+            length (int): Length of the dataset.
+        """
+        return self.length
+
+    def __getitem__(self, i):
+        """Iterator for rollout sampling.
+        Samples a rollout and converts it to a Pytorch tensor.
+
+        Args:
+            i (int): Index of the dataset sample (ignored since we sample random data).
+        """
+        rolls = self.environment.sample_random_rollouts(number_of_frames=self.number_of_frames,
+                                                        delta_time=self.delta_time,
+                                                        number_of_rollouts=self.number_of_rollouts,
+                                                        img_size=self.img_size,
+                                                        noise_std=self.noise_std,
+                                                        radius_bound=self.radius_bound,
+                                                        world_size=self.world_size,
+                                                        seed=self.seed)
+        # standarization
+        rolls = (rolls - self.data_mean)/self.data_std
```

Comment:
Arent the rolls from 0-255? I don't understand how you are normalizing here

Reply:
the rolls are [0 ... 1]